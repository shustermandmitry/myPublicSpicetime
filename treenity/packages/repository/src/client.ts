export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
