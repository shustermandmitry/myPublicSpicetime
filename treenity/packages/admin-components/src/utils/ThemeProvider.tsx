/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
