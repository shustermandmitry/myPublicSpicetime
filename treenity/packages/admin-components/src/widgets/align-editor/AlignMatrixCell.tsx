/*
 * Copyright (c) 2024. Treenity Inc.
 */

import Icon from '@/components/icon';
import useHover from '@/hooks/useHover';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { omitProps } from '@treenity/ui-kit/utils';
import { FC } from 'react';
import type {
  AlignCellsType,
  AlignDisplayType,
  AlignOrientationProps,
  IAlignIconProps,
  IAlignMatrixCell,
} from './types';

const row: AlignDisplayType = {
  center_start: 'top-row_outlined',
  left_start: 'top-row_outlined',
  right_start: 'top-row_outlined',
  left_center: 'center-row_outlined',
  center_center: 'center-row_outlined',
  right_center: 'center-row_outlined',
  center_end: 'bottom-row_outlined',
  left_end: 'bottom-row_outlined',
  right_end: 'bottom-row_outlined',
  'space-between_start': 'row-top-space-between_outlined',
  'space-between_center': 'row-center-space-between_outlined',
  'space-between_end': 'row-bottom-space-between_outlined',
  'space-between_stretch': 'space-between_outlined',
  'space-around_start': 'row-top-space-around_outlined',
  'space-around_center': 'row-center-space-around_outlined',
  'space-around_end': 'row-bottom-space-around_outlined',
  'space-around_stretch': 'space-around_outlined',
  center_stretch: 'stretch-horizontal_outlined',
  left_stretch: 'stretch-horizontal_outlined',
  right_stretch: 'stretch-horizontal_outlined',
};

const GeneralAlignIcons: AlignDisplayType = {
  left_start: 'left-column_outlined',
  center_center: 'center-column_outlined',
  right_end: 'right-column_outlined',
  'space-between_start': 'column-bottom-space-between_outlined',
  'space-between_center': 'column-center-space-between_outlined',
  'space-between_end': 'column-top-space-between_outlined',
  'space-between_stretch': 'space-between-vertical_outlined',
  'space-around_start': 'column-bottom-space-around_outlined',
  'space-around_center': 'column-center-space-around_outlined',
  'space-around_end': 'column-top-space-around_outlined',
  'space-around_stretch': 'space-around-vertical_outlined',
  center_stretch: 'stretch-vertical_outlined',
  left_stretch: 'stretch-vertical_outlined',
  right_stretch: 'stretch-vertical_outlined',
};

export const alignIcons: IAlignIconProps = {
  grid: {
    column: {
      left_center: 'left-column_outlined',
      left_end: 'left-column_outlined',
      center_start: 'center-column_outlined',
      center_end: 'center-column_outlined',
      right_start: 'right-column_outlined',
      right_center: 'right-column_outlined',
      ...GeneralAlignIcons,
    },
    row,
  },
  flex: {
    column: {
      left_center: 'center-column_outlined',
      left_end: 'right-column_outlined',
      center_start: 'left-column_outlined',
      center_end: 'right-column_outlined',
      right_start: 'left-column_outlined',
      right_center: 'center-column_outlined',
      ...GeneralAlignIcons,
    },
    row,
  },
};

const MatrixPointIcon = <Icon name="matrix-point_outlined" color="gray700" />;

const AlignMatrixCell: FC<IAlignMatrixCell> = ({
  value,
  orientation,
  selected,
  cellsType,
  onClick,
  display,
}) => {
  const [refCell, hovered] = useHover();
  const iconName = alignIcons[display as keyof IAlignIconProps][orientation][value];

  return (
    <Cell
      ref={refCell}
      onClick={onClick}
      cellsType={cellsType}
      orientation={orientation}
      about={value}
    >
      {iconName &&
        (hovered || selected ? (
          <IconStyled
            name={iconName}
            color="primary"
            cellsType={cellsType}
            orientation={orientation}
          />
        ) : cellsType === 'full' ? (
          MatrixPointIcon
        ) : (
          cellsType === 'triple' && [MatrixPointIcon, MatrixPointIcon, MatrixPointIcon]
        ))}
    </Cell>
  );
};

const IconStyled = styled(Icon, omitProps('cellsType', 'orientation'))<{
  cellsType: AlignCellsType;
  orientation: AlignOrientationProps;
}>`

  && i {
    font-size: ${p =>
      (p.orientation === 'column' && p.cellsType === 'triple') || p.cellsType === 'single'
        ? 54
        : 18}px
`;

const Cell = styled('div', omitProps('cellsType', 'orientation'))<{
  cellsType: AlignCellsType;
  orientation: AlignOrientationProps;
}>`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  color: ${p => p.theme.colorBorder};

  ${p =>
    p.cellsType === 'triple' &&
    css`
      justify-content: space-around;
    `}

  ${p =>
    p.orientation === 'column' &&
    css`
      flex-direction: column;
    `};

  :where(i) {
    color: ${p => p.theme.colorPrimary};
  }
`;

export default AlignMatrixCell;
