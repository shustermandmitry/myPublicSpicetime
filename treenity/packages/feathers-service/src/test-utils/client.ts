import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
