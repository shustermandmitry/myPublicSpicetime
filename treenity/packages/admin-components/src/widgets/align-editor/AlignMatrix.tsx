/*
 * Copyright (c) 2024. Treenity Inc.
 */

import styled from '@emotion/styled';
import { omitProps } from '@treenity/ui-kit/utils';
import { FC, useMemo } from 'react';
import AlignMatrixCell from './AlignMatrixCell';
import { AlignCellsType, AlignMatrixProps, AlignOrientationProps, AlignVariation } from './types';

export const getValue = (index: number, value?: AlignVariation) => {
  if (value) {
    return value.split('_')[index];
  }
  return null;
};

const cellsFull: AlignVariation[] = [
  'left_start',
  'center_start',
  'right_start',
  'left_center',
  'center_center',
  'right_center',
  'left_end',
  'center_end',
  'right_end',
];

const cellsFlexColumn: AlignVariation[] = [
  'left_start',
  'left_center',
  'left_end',
  'center_start',
  'center_center',
  'center_end',
  'right_start',
  'right_center',
  'right_end',
];

const AlignMatrix: FC<AlignMatrixProps> = ({
  value = 'left_start',
  orientation = 'row',
  onSelect,
  display,
}) => {
  const [firstValue, secondValue] = value.split('_');

  const cells: {
    full(): AlignVariation[];
    triple(value: string): AlignVariation[];
    single(value: string): AlignVariation[];
  } = useMemo(
    () => ({
      full: () => {
        if (display === 'flex' && orientation === 'column') {
          return cellsFlexColumn;
        } else {
          return cellsFull;
        }
      },
      triple: (value: string) =>
        value === 'stretch'
          ? ([`left_${value}`, `center_${value}`, `right_${value}`] as AlignVariation[])
          : ([`${value}_start`, `${value}_center`, `${value}_end`] as AlignVariation[]),
      single: (value: string) => [`${value}_stretch`] as AlignVariation[],
    }),
    [display, orientation],
  );

  const cellsType = useMemo(() => {
    if (value === 'space-around_stretch' || value === 'space-between_stretch') {
      return 'single';
    } else if (
      firstValue === 'space-between' ||
      firstValue === 'space-around' ||
      secondValue === 'stretch'
    ) {
      return 'triple';
    }
    return 'full';
  }, [value, firstValue, secondValue]);

  const valueIndex = useMemo(() => {
    if (firstValue !== 'space-between' && firstValue !== 'space-around') {
      if (secondValue === 'stretch') {
        return 1;
      }
    }
    return 0;
  }, [firstValue, secondValue]);

  return (
    <Root cellsType={cellsType} orientation={orientation}>
      {cells[cellsType](getValue(valueIndex, value) as string).map((cell, index) => (
        <AlignMatrixCell
          display={display}
          key={index}
          value={cell}
          orientation={orientation}
          selected={cell === value}
          cellsType={cellsType}
          onClick={() => onSelect(cell)}
        />
      ))}
    </Root>
  );
};

const Root = styled('div', omitProps('cellsType', 'orientation'))<{
  cellsType: AlignCellsType;
  orientation: AlignOrientationProps;
}>`
  width: 60px;
  height: 60px;
  display: grid;
  grid-template-columns: ${p =>
    p.cellsType === 'triple' && p.orientation === 'row' ? '1fr' : 'repeat(3, 1fr)'};
  grid-template-rows: ${p =>
    p.cellsType === 'triple' && p.orientation === 'column' ? '1fr' : 'repeat(3, 1fr)'};
  border: 1px solid ${p => p.theme.colorBorder};
  background-color: ${p => p.theme.colorBgContainer};
  border-radius: 6px;
  padding: 2px;
`;

export default AlignMatrix;
