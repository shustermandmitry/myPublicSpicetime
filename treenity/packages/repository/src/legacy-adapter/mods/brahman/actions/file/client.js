import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
