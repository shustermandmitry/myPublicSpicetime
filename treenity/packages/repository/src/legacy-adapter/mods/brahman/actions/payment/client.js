import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
