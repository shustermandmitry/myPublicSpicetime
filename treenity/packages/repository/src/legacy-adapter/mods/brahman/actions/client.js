import './addActionMenu';
import './back/TelegramActionBack.client';
import './broadcast/TelegramActionBroadcast.client';

import './call-command-by-param/TGActionCallCommandByParam.client';
import './crypto/TelegramActionGetRates.client';
// import './voice/video/SynhesizeAudioVideo';
import './emit-text/EmitText.client';
import './file/TelegramActionFileSend.client';
import './file/TelegramActionFileUpload.client';
import './file/TelegramFile.client';
import './forward/TelegramActionForward.client';
import './game/CurrentGame.client';
import './game/Game.client';
import './game/TelegramActionGame.client';
import './get-value/GetValue.client';
import './i18n/TGActionSelectLanguage.client';
import './if-else/TelegramActionIfElse.client';

import './keyword-select/TelegramActionKeywordSelect.client';
import './menu/TelegramActionMenu.client';
import './message/TelegramActionMessage.client';
import './on-error/TelegramActionOnError.client';
import './page/TelegramActionPage.client';
// import './pantaro/client';
import './params/TelegramActionParams.client';
import './payment/client';
import './question/TelegramActionQuestion.client';
import './removeMessage/TelegramActionRemove.client';
import './reset-history/ResetSession.client';
import './reset-session/ResetSession.client';
import './set-telegram-user/TGActionSetTelegramUser.client';
import './set-value/SetValue.client';

import './supply-courses/client';

import './tag/SetTag';
import './TelegramAction.client';
import './timers/TgActionSetTimer';

import './timers/timers.service.client';

import './translate/TString.client';
import './voice/tinkoff/TinkoffVoiceRecognition.client';
import './voice/tinkoff/TinkoffVoiceSynthesize.client';

import './voice/yandex/YandexVoiceRecognition.client';
import './voice/yandex/YandexVoiceSynthesize.client';

import './ym-statistics/TelegramPageStatistics.client';
