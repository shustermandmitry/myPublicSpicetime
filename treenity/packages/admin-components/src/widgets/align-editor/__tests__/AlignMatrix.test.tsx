/*
 * Copyright (c) 2024. Treenity Inc.
 */
import ThemeProvider from '@/utils/ThemeProvider';
import { jest } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';
import { CookieContext, createCookies, parseCookies, setCookie } from '@treenity/repository/client';
import AlignMatrix from '../AlignMatrix';
import { alignIcons } from '../AlignMatrixCell';
import type { AlignOrientationProps, AlignVariation } from '../types';

const cookies = createCookies(parseCookies(document.cookie), setCookie);

const defaultValue: AlignVariation | undefined = undefined;

let valueMock: AlignVariation | undefined = defaultValue;

const updateMock = jest.fn((value: AlignVariation) => {
  valueMock = value;
  return value;
});

const orientation: AlignOrientationProps[] = ['row', 'column'];

describe('AlignMatrix', () => {
  beforeEach(() => {
    updateMock.mockClear();
    valueMock = defaultValue;
  });

  it('render AlignMatrix component', async () => {
    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[0]}
            onSelect={updateMock}
            value={valueMock}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const items = root!.querySelectorAll(`body > div > div > div`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(items).toHaveLength(9);

    [...items].forEach((item, index) => {
      const icon = item.querySelector(`i`);
      expect(item).toBeInTheDocument();
      expect(icon).toBeInTheDocument();
      expect(icon).toHaveClass('icon-matrix-point_outlined');
    });
  });

  it('render AlignMatrix component with triple value', async () => {
    const newValue = 'left_stretch';

    const { container, rerender } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[0]}
            onSelect={updateMock}
            value={newValue}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    expect(root).toBeInTheDocument();
    const items = root!.querySelectorAll(`body > div > div > div`);

    expect(root).toMatchSnapshot();
    expect(items).toHaveLength(3);

    [...items].forEach((item, index) => {
      const icons = item.querySelectorAll(`i`);
      expect(item).toBeInTheDocument();

      if (index === 0) {
        expect(icons).toHaveLength(1);
        expect(icons[0]).toBeInTheDocument();
        expect(icons[0]).toHaveClass('icon-stretch-horizontal_outlined');
      } else {
        [...icons].forEach(icon => {
          expect(icons).toHaveLength(3);
          expect(icon).toBeInTheDocument();
          expect(icon).toHaveClass('icon-matrix-point_outlined');
        });
      }
    });

    fireEvent.click(items[1]);

    rerender(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[0]}
            onSelect={updateMock}
            value={valueMock}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(updateMock.mock.calls).toHaveLength(1);

    expect(updateMock.mock.calls.at(-1)).toEqual(['center_stretch']);

    const items2 = root!.querySelectorAll(`body > div > div > div`);
    expect(items2[1]).toBeInTheDocument();
    const icon = items2[1]!.querySelector(`i`);
    expect(icon).toBeInTheDocument();
    expect(icon).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon).toHaveClass(`icon-${alignIcons['flex'][orientation[0]][valueMock!]}`);
  });

  it('render AlignMatrix component with triple and origin column', async () => {
    const newValue = 'left_stretch';

    const { container, rerender } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[1]}
            onSelect={updateMock}
            value={newValue}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    expect(root).toBeInTheDocument();
    const items = root!.querySelectorAll(`body > div > div > div`);

    expect(root).toMatchSnapshot();
    expect(items).toHaveLength(3);

    [...items].forEach((item, index) => {
      const icons = item.querySelectorAll(`i`);
      expect(item).toBeInTheDocument();

      if (index === 0) {
        expect(icons).toHaveLength(1);
        expect(icons[0]).toBeInTheDocument();
        expect(icons[0]).toHaveClass('icon-stretch-vertical_outlined');
      } else {
        [...icons].forEach(icon => {
          expect(icons).toHaveLength(3);
          expect(icon).toBeInTheDocument();
          expect(icon).toHaveClass('icon-matrix-point_outlined');
        });
      }
    });

    fireEvent.click(items[1]);

    rerender(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[0]}
            onSelect={updateMock}
            value={valueMock}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(updateMock.mock.calls).toHaveLength(1);

    expect(updateMock.mock.calls.at(-1)).toEqual(['center_stretch']);

    const items2 = root!.querySelectorAll(`body > div > div > div`);
    expect(items2[1]).toBeInTheDocument();
    const icon = items2[1]!.querySelector(`i`);
    expect(icon).toBeInTheDocument();
    expect(icon).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon).toHaveClass(`icon-${alignIcons['flex'][orientation[0]][valueMock!]}`);
  });

  it('render AlignMatrix component with single value', async () => {
    const newValue = 'space-around_stretch';

    const { container, rerender } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[1]}
            onSelect={updateMock}
            value={newValue}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    expect(root).toBeInTheDocument();
    const items = root!.querySelectorAll(`body > div > div > div`);

    expect(root).toMatchSnapshot();
    expect(items).toHaveLength(1);

    [...items].forEach((item, index) => {
      const icons = item.querySelectorAll(`i`);
      expect(item).toBeInTheDocument();

      if (index === 0) {
        expect(icons).toHaveLength(1);
        expect(icons[0]).toBeInTheDocument();
        expect(icons[0]).toHaveClass('icon-space-around-vertical_outlined');
      } else {
        [...icons].forEach(icon => {
          expect(icons).toHaveLength(3);
          expect(icon).toBeInTheDocument();
          expect(icon).toHaveClass('icon-matrix-point_outlined');
        });
      }
    });

    fireEvent.click(items[0]);

    rerender(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrix
            orientation={orientation[0]}
            onSelect={updateMock}
            value={valueMock}
            display={'flex'}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(updateMock.mock.calls).toHaveLength(1);
    expect(updateMock.mock.calls.at(-1)).toEqual(['space-around_stretch']);
    const items2 = root!.querySelectorAll(`body > div > div > div`);
    expect(items2).toHaveLength(1);
    expect(items2[0]).toBeInTheDocument();
    const icon = items2[0]!.querySelector(`i`);
    expect(icon).toBeInTheDocument();
    expect(icon).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon).toHaveClass(`icon-${alignIcons['flex'][orientation[0]][valueMock!]}`);
  });
});
