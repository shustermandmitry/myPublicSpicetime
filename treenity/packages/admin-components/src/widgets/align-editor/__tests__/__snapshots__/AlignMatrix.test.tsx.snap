// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignMatrix render AlignMatrix component 1`] = `
.emotion-0 {
  width: 60px;
  height: 60px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  border: 1px solid #d9d9d9;
  background-color: #ffffff;
  border-radius: 6px;
  padding: 2px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
}

.emotion-1:where(i) {
  color: #1677ff;
}

.emotion-2 [class^='icon-'],
.emotion-2 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-3 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: gray700;
}

<div
  class="emotion-0"
>
  <div
    about="left_start"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="center_start"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="right_start"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="left_center"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="center_center"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="right_center"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="left_end"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="center_end"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="right_end"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-2"
    >
      <i
        class="icon-matrix-point_outlined emotion-3"
      />
    </span>
  </div>
</div>
`;

exports[`AlignMatrix render AlignMatrix component with single value 1`] = `
.emotion-0 {
  width: 60px;
  height: 60px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  border: 1px solid #d9d9d9;
  background-color: #ffffff;
  border-radius: 6px;
  padding: 2px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1:where(i) {
  color: #1677ff;
}

.emotion-2 [class^='icon-'],
.emotion-2 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-2.emotion-2 i {
  font-size: 54px;
}

.emotion-3 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #1677ff;
}

<div
  class="emotion-0"
>
  <div
    about="space-around_stretch"
    class="emotion-1"
  >
    <span
      class="anticon emotion-2"
    >
      <i
        class="icon-space-around-vertical_outlined emotion-3"
      />
    </span>
  </div>
</div>
`;

exports[`AlignMatrix render AlignMatrix component with triple and origin column 1`] = `
.emotion-0 {
  width: 60px;
  height: 60px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  border: 1px solid #d9d9d9;
  background-color: #ffffff;
  border-radius: 6px;
  padding: 2px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1:where(i) {
  color: #1677ff;
}

.emotion-2 [class^='icon-'],
.emotion-2 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-2.emotion-2 i {
  font-size: 54px;
}

.emotion-3 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #1677ff;
}

.emotion-5 [class^='icon-'],
.emotion-5 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-6 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: gray700;
}

<div
  class="emotion-0"
>
  <div
    about="left_stretch"
    class="emotion-1"
  >
    <span
      class="anticon emotion-2"
    >
      <i
        class="icon-stretch-vertical_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="center_stretch"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
  </div>
  <div
    about="right_stretch"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
  </div>
</div>
`;

exports[`AlignMatrix render AlignMatrix component with triple value 1`] = `
.emotion-0 {
  width: 60px;
  height: 60px;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: repeat(3, 1fr);
  border: 1px solid #d9d9d9;
  background-color: #ffffff;
  border-radius: 6px;
  padding: 2px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-1:where(i) {
  color: #1677ff;
}

.emotion-2 [class^='icon-'],
.emotion-2 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-2.emotion-2 i {
  font-size: 18px;
}

.emotion-3 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #1677ff;
}

.emotion-5 [class^='icon-'],
.emotion-5 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-6 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: gray700;
}

<div
  class="emotion-0"
>
  <div
    about="left_stretch"
    class="emotion-1"
  >
    <span
      class="anticon emotion-2"
    >
      <i
        class="icon-stretch-horizontal_outlined emotion-3"
      />
    </span>
  </div>
  <div
    about="center_stretch"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
  </div>
  <div
    about="right_stretch"
    class="emotion-1"
  >
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
    <span
      class="anticon undefined emotion-5"
    >
      <i
        class="icon-matrix-point_outlined emotion-6"
      />
    </span>
  </div>
</div>
`;
